src/NodeAnalyzer/DependencyInjectionMethodCallAnalyzer.php

```diff
@@ -38,10 +43,32 @@ public function replaceMethodCallWithPropertyFetchAndDependency(MethodCall $meth
         }
 
         $propertyName = $this->propertyNaming->fqnToVariableName($serviceType);
+        $propertyName = $this->resolveNewPropertyNameWhenExists($class, $propertyName);
 
         $propertyMetadata = new PropertyMetadata($propertyName, $serviceType, Class_::MODIFIER_PRIVATE);
         $this->propertyToAddCollector->addPropertyToClass($class, $propertyMetadata);
 
         return $this->nodeFactory->createPropertyFetch('this', $propertyName);
     }
+
+    private function resolveNewPropertyNameWhenExists(Class_ $class, string $propertyName, int $count = 1): string
+    {
+        $promotedPropertyParams = $this->promotedPropertyResolver->resolveFromClass($class);
+        foreach ($promotedPropertyParams as $promotedPropertyParam) {
+            if ($this->nodeNameResolver->isName($promotedPropertyParam->var, $propertyName)) {
+                ++$count;
+                $propertyName .= $count;
+                return $this->resolveNewPropertyNameWhenExists($class, $propertyName, $count);
```

Comment:
current `VariableNaming` count++ naming only handle variable, so use recursive check here to check property exists.